cms/utils/placeholder.py

```diff
@@ -363,3 +365,25 @@ def get_declared_placeholders_for_obj(obj):
     if not hasattr(obj, 'get_template'):
         raise NotImplementedError('%s should implement get_template' % obj.__class__.__name__)
     return get_placeholders(obj.get_template())
+
+
+def run_placeholder_container_checks(container, user):
+    """
+    Get all of the placeholders from the parents and
```

Comment:
all of placeholders linked to container? I'd also rename container to source here, as that's the name used in a bunch of places already